### PR TITLE
[HZ-4181] Deprecate Portable

### PR DESCRIPTION
**Pull request by avtarraikmo**
Hi team - I need some help here.

1. Mark Portable as Deprecated
2. Update the rationale for Compact Serialization

Can you help me sharpen this to be effective?

### CODE REVIEW

File: docs/modules/serialization/pages/serialization.adoc [hunk 1]
```diff
@@ -58,6 +59,18 @@ may incur costs on short-lived use cases
...
 
+|All clients
+
+| xref:custom-serialization.adoc[Custom Serialization]
+| * Does not require class to implement an interface
+
+* Convenient and flexible
+
+* Can be based on `StreamSerializer` `ByteArraySerializer`
+|* Serialization interface must be implemented
```
Comment (rebekah-lawrence) [inline docs/modules/serialization/pages/serialization.adoc:57]: I'd suggest "Must implement serialization interface", however, there are multiple uses of this phrase on the page so either change all or none

File: docs/modules/serialization/pages/serialization.adoc [hunk 2]
```diff
@@ -58,6 +59,18 @@ may incur costs on short-lived use cases
...
+
+* Plug in and configuration is required
```
Comment (rebekah-lawrence) [inline docs/modules/serialization/pages/serialization.adoc:59]:
```suggestion
* Requires plug-in and configuration
```

File: docs/modules/serialization/pages/serialization.adoc [hunk 3]
```diff
@@ -68,6 +68,25 @@ may incur costs on short-lived use cases
 |===
 | Serialization interface| Advantages| Disadvantages|Client support
 
+| xref:implementing-dataserializable.adoc[`IdentifiedDataSerializable`]
+| * More CPU and memory usage efficient than `Serializable`
```
Comment (rebekah-lawrence) [inline docs/modules/serialization/pages/serialization.adoc:72]:
```suggestion
| * More efficient CPU and memory usage than `Serializable`
```

File: docs/modules/serialization/pages/serialization.adoc [hunk 4]
```diff
@@ -21,7 +21,8 @@ Hazelcast offers the following serialization options:
 | Serialization interface| Advantages| Disadvantages|Client support
 
 | xref:compact-serialization.adoc[Compact Serialization]
-| * More memory usage efficient than `Portable`
+Recommended
+| * Memory usage and efficiency optimized format
```
Comment (oliverhowell) [inline docs/modules/serialization/pages/serialization.adoc:25]:
```suggestion
| * Optimized for memory usage
```

File: docs/modules/serialization/pages/serialization.adoc [hunk 5]
```diff
@@ -21,7 +21,8 @@ Hazelcast offers the following serialization options:
 | Serialization interface| Advantages| Disadvantages|Client support
 
 | xref:compact-serialization.adoc[Compact Serialization]
-| * More memory usage efficient than `Portable`
+Recommended
```
Comment (oliverhowell) [inline docs/modules/serialization/pages/serialization.adoc:24]: Maybe needs either some formatting or expanded into a sentence e.g. Recommended for...

File: docs/modules/serialization/pages/serialization.adoc [hunk 6]
```diff
@@ -68,6 +68,25 @@ may incur costs on short-lived use cases
...
 | Serialization interface| Advantages| Disadvantages|Client support
 
+| xref:implementing-dataserializable.adoc[`IdentifiedDataSerializable`]
+| * More CPU and memory usage efficient than `Serializable`
+
+* Reflection is not used during deserialization
+
+| * Specific to Hazelcast
+
+* Serialization interface must be implemented
```
Comment (rebekah-lawrence) [inline docs/modules/serialization/pages/serialization.adoc:78]:
```suggestion
* Must implement serialization interface
```

File: docs/modules/serialization/pages/serialization.adoc [hunk 7]
```diff
@@ -68,6 +68,25 @@ may incur costs on short-lived use cases
...
+
+* A factory and configuration must be implemented
```
Comment (rebekah-lawrence) [inline docs/modules/serialization/pages/serialization.adoc:80]:
```suggestion
* Must implement and configure a factory
```

File: docs/modules/serialization/pages/serialization.adoc [hunk 8]
```diff
@@ -68,6 +68,25 @@ may incur costs on short-lived use cases
...
+
+|All clients
+
+| xref:implementing-dataserializable.adoc[`DataSerializable`]
+| * More CPU and memory usage efficient than `Serializable`
```
Comment (rebekah-lawrence) [inline docs/modules/serialization/pages/serialization.adoc:85]:
```suggestion
| * More efficient CPU and memory usage than `Serializable`
```

File: docs/modules/serialization/pages/serialization.adoc (hunk 7 above)
Comment (oliverhowell) [inline docs/modules/serialization/pages/serialization.adoc:80]: What is a factory? I see "A Factory and configuration must be implemented" used elsewhere on page

File: docs/modules/serialization/pages/serialization.adoc [hunk 9]
```diff
@@ -86,13 +105,8 @@ may incur costs on short-lived use cases
...
 |Java only
 
-| xref:implementing-dataserializable.adoc[`DataSerializable`]
-| * More CPU and memory usage efficient than `Serializable`
-| * Specific to Hazelcast
-
-|Java only
-
 | xref:implementing-portable-serialization.adoc[`Portable`]
+*Deprecated: Use Compact Serialization. This will be removed as of version 6.0*
```
Comment (rebekah-lawrence) [inline docs/modules/serialization/pages/serialization.adoc:109]:
```suggestion
*Deprecated: Use Compact Serialization. Portable Serialization will be removed as of version 6.0*
```

Comment (oliverhowell) [inline docs/modules/serialization/pages/serialization.adoc:109]: Can we turn this into a admonition or warning? Not familiar enough with the syntax yet but drawing attention to it with colour/symbol etc. would help.

I'd go for a "Hazelcast strongly recommends using Compact Serialization as Portable will be deprecated and removed in version 6.0" or similar...

**Review by oliverhowell — COMMENTED** [review]
Some hopefully helpful quick comments...

**Review by Serdaro — COMMENTED** [review]
LGTM after recent updates.

**Comment by Serdaro** [comment]
@oliverhowell , I am seeing the same deprecation note twice in a fairly small page (serialization.adoc)

**Comment by oliverhowell** [comment]
> @oliverhowell , I am seeing the same deprecation note twice in a fairly small page (serialization.adoc). You can see the look and feel from the link listed under Netlify bot's comment above, which is the staging site (the "Deploy Preview" link).

Good spot! It was partly deliberate but the second note was _meant_ to be inside the collapsed section. I've pushed a change to put it where it was meant to be. Will see how this version displays but I thought it was worth repeating as the second Alternate section has Portable listed and it's quite far from the original deprecation notice (there were two original mentions in the original PR too but I've turned them both into proper Notes so they are more visible). Let me know what you think of new version and whether the second Note is needed or not.

File: docs/modules/serialization/pages/implementing-portable-serialization.adoc [hunk 10]
```diff
@@ -1,5 +1,7 @@
 = Implementing Portable Serialization
 
+#NOTE: Portable Serialization has been deprecated. Use Compact Serialization. Portable Serialization will be removed as of version 6.0#
```
[the remark this answers is not in the document]
Comment (oliverhowell) [reply docs/modules/serialization/pages/implementing-portable-serialization.adoc]: Implemented

File: docs/modules/serialization/pages/serialization.adoc [hunk 11]
```diff
@@ -21,11 +21,12 @@ Hazelcast offers the following serialization options:
 | Serialization interface| Advantages| Disadvantages|Client support
 
 | xref:compact-serialization.adoc[Compact Serialization]
-| * More memory usage efficient than `Portable`
+#Recommended#
```
[the remark this answers is not in the document]
Comment (oliverhowell) [reply docs/modules/serialization/pages/serialization.adoc]: Implemented

File: docs/modules/serialization/pages/serialization.adoc [hunk 12]
```diff
@@ -47,16 +48,15 @@ may incur costs on short-lived use cases
...
-* Reflection is not used during deserialization
+| xref:custom-serialization.adoc[Custom Serialization]
+| * Does not require class to implement an interface
 
-| * Specific to Hazelcast
+* Convenient and flexible
 
-* Serialization interface must be implemented
+* Can be based on `StreamSerializer` or `ByteArraySerializer`
+|* Serialization interface must be implemented
```
[the remark this answers is not in the document]
Comment (oliverhowell) [reply docs/modules/serialization/pages/serialization.adoc]: Implemented

**Review by oliverhowell — APPROVED** [review]
Recommendations and changes made in my last commit so can ignore other comments - just needs a sanity/double check on confirming 7.0 as release it will be removed but I'm good with latest version so will approve from my side.

**Comment by avtarraikmo** [comment]
Looks good to me.

**Review by rebekah-lawrence — APPROVED** [review]
LGTM

**Pull request merged**